tests/test_database_database.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import pytest  # type: ignore
+import tempfile
+
+from nixnet import database
+
+
+@pytest.mark.integration
+def test_database_database_container():
+    with database.Database('NIXNET_example') as db_1:
+        with database.Database('NIXNET_exampleLDF') as db_2:
+            db_3 = database.Database('NIXNET_example')
+
+            # test dunders
+            assert db_1 == db_3
+            assert db_1 != db_2
+            assert len({db_1, db_2, db_3}) == 2  # testing `__hash__`
+            print(repr(db_1))
+
+            db_3.close()
+            del db_3
```

Comment:
good call on explicitly calling `close` here. we call `close` on object exit but I don't think that call is deterministic....it might depend on the python runtime behavior.

in general, I think we should explicitly close databases that were not created using a with-statement. let me know if have thoughts/comments otherwise.

Reply:
`db_1` and `db_2` will indeed be closed for the user by the API when the `__exit__` dunder for the with-statement executes.

Conversely, for `db_3`, `close` will not be called automatically for the user. The user must manually call `close`. If the user doesn't call `close`, the API detects that in the `__del__` destructor dunder during garbage collection and emits a warning to the user (but does not close the database).

Yes, I agree that users should be calling `close` on Database instances that were not created using a with-statement.